From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tavi <tavi@divested.dev>
Date: Fri, 2 Jul 2021 13:03:00 -0400
Subject: [PATCH 8/9] build: hardening: misc added flags

Signed-off-by: Tavi <tavi@divested.dev>
---
 include/hardening.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/hardening.mk b/include/hardening.mk
index 1565e5aa27..7f8afd5e1a 100644
--- a/include/hardening.mk
+++ b/include/hardening.mk
@@ -77,3 +77,4 @@ ifdef CONFIG_PKG_DT_RELR
   endif
 endif
 
+TARGET_CFLAGS += -fstack-clash-protection -fwrapv
